backend/fms_core/services/derived_sample.py

```diff
@@ -14,4 +14,23 @@ def inherit_derived_sample(derived_sample_source, new_derived_sample_data):
     except Error as e:
         errors.append(';'.join(e.messages))
     
-    return (new_derived_sample, errors, warnings)
\ No newline at end of file
+    return (new_derived_sample, errors, warnings)
+
+def update_qc_flags(sample, quantity_flag, quality_flag):
+    errors = []
+    warnings = []
+
+    try:
+        derived_sample = sample.derived_sample_not_pool
+        if derived_sample:
+            if quantity_flag and quality_flag:
+                derived_sample.quantity_flag = True if quantity_flag == 'Passed' else False
```

Comment:
you could also used the simplified : derived_sample.quantity_flag = (quantity_flag == 'Passed').